### PR TITLE
Use repo's root as the `proto_path` in calls to `compile_proto` 

### LINKED ISSUE #2167: Use the root repo as the proto_path in all invocations of prost
Ref discussion on Slack: https://project-oak.slack.com/archives/CHE9E13C3/p1622730588061800

### PR DESCRIPTION
**Pull request by rbehjati**
Fixes #2167

The recommended approach for using `protoc` is to use the root of the source tree as the proto path, instead of including multiple paths. All paths must be specified relative to the root of the source tree. This change list updates all invocations of `compile_protos` and `generate_grpc_code` according to this advice. 

I have updated the documentation of `compile_protos` and `generate_grpc_code`, but I don't think there is a way to actually enforce this.

### CODE REVIEW

**Review by conradgrobler — APPROVED** [review]
Nice cleanup.

**Comment by rbehjati** [comment]
I'll merge this after #2156 is merged.

**Review by ipetr0v — APPROVED** [review]

**Pull request merged**
